layout-server/src/main/java/org/finos/vuu/layoutserver/service/ApplicationLayoutService.java

```diff
@@ -0,0 +1,76 @@
+package org.finos.vuu.layoutserver.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.finos.vuu.layoutserver.model.ApplicationLayout;
+import org.finos.vuu.layoutserver.repository.ApplicationLayoutRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class ApplicationLayoutService {
+
+    private static final Logger logger = LoggerFactory.getLogger(ApplicationLayoutService.class);
+    private static final String DEFAULT_LAYOUT_FILE = "defaultLayout.json";
+    private static ApplicationLayout defaultLayout;
+    private final ApplicationLayoutRepository repository;
+
+    public void createApplicationLayout(String username, JsonNode layoutDefinition) {
+        repository.save(new ApplicationLayout(username, layoutDefinition));
+    }
+
+    public ApplicationLayout getApplicationLayout(String username) {
+        Optional<ApplicationLayout> layout = repository.findById(username);
```

Comment:
Discussed offline: this is fine for `findByUsername()`, but we'd then want to do the same with `deleteByUsername()`. This is possible, but it would not throw exceptions in the same way. Decided to keep it as is, because username is actually the ID at present. This may change in future when users are fully implemented/persisted.